plumgrid_playbooks/roles/plumgrid_director/templates/default.conf

```diff
@@ -12,6 +12,14 @@ upstream pgCli {
   server {{ nginx_virtual_ip }}:{{ pg_ports['pgcli_port'] }};
 }
 
+upstream pgMW {
+{% if middleware_ip is defined %}
+  server {{ middleware_ip }}:{{ pg_ports['mw_port'] }};
```

Comment:
Why not always have a jinja variable for middleware ip which defaults to 127.0.0.1
Would save you an if else statement over here.